server/src/main/java/org/elasticsearch/index/fielddata/MultiGeoValues.java

```diff
@@ -205,14 +220,17 @@ public static GeoShapeValue missing(String missing) {
     }
 
     public static class BoundingBox {
-        public final double top;
-        public final double bottom;
-        public final double negLeft;
-        public final double negRight;
-        public final double posLeft;
-        public final double posRight;
-
-        public BoundingBox(Extent extent, CoordinateEncoder coordinateEncoder) {
```

Comment:
I wonder how much this really buys us performance-wise in modern JVM? Did you run some benchmarks by any chance?

Reply:
I did some performance testing in a throw-away-branch (independent of this) where I limited the creation of Extents and such. I didn't see a significant performance difference, just a difference in object allocations which may help reduce garbage collection time.

Reply:
These are all short lived objects. So, my gut feel it shouldn't be bad

Reply:
true. it was less significant for when there was one BoundingBox per document/field-value.

The more important place where there is significant churn is in the geo-grid tiler that creates a new Rectangle for every tile queried. There is where I did not see a performance difference, just an object allocation difference

Reply:
Throughput of the query should not be much different but latency can suffer if we create millions of this objects as gc will play a role. This code runs on a tight loop and you can have several of this loops running in parallel so creating objects per document is a bad idea.